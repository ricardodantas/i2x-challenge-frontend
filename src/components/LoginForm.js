import React, { Component } from 'react'
import PropTypes from 'prop-types'
// import cssModules from 'react-css-modules'
// import styles from '../style/counter.scss'
import { Button, Form, Input, Container, Message } from 'semantic-ui-react'
import _ from 'lodash'

// @cssModules(styles)

export default class LoginForm extends Component {

  static propTypes = {
    // styles: PropTypes.object,
    handleChange: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    response: PropTypes.object,
  }

  render() {
    const { response, handleChange, handleSubmit} = this.props

    return (
      <Container className="ui raised very padded text container segment">
        <h2>Login</h2>
        <Form onSubmit={handleSubmit}>
          <Form.Field>
            <Input type="email" name="email" placeholder='email' onChange={handleChange}/>
          </Form.Field>
          <Form.Field>
            <Input type="password" name="password" placeholder='password' onChange={handleChange}/>
          </Form.Field>
          <Button primary={true} type='submit'>Sign In</Button>
        </Form>
        <Message negative hidden={!response.hasFailed}>
          <Message.Header>Ops...</Message.Header>
          <p>{response.response ? response.response.message : 'Something is wrong.'}</p>
        </Message>
      </Container>
    )
  }
}
