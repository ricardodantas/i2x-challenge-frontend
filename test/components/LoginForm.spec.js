import React from 'react'
import { spy } from 'sinon'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import LoginForm from '../../src/components/LoginForm'

describe('<LoginForm />', () => {

  beforeEach(() => {

    global.handleChange = spy()
    global.handleSubmit = spy()
    global.response = {
      response: { }
    }
  })

  it('should render', () => {
    const tree = renderer.create(
      <LoginForm
        handleChange={global.handleChange}
        handleSubmit={global.handleSubmit}
        response={global.response}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should call handleSubmit when form is submited', () => {
    const component = shallow(
      <LoginForm
        handleChange={global.handleChange}
        handleSubmit={global.handleSubmit}
        response={global.response}
      />
    )

    const form = component.find('Form')
    form.simulate('submit')

    expect(global.handleSubmit.calledOnce).toEqual(true)
  })

  it('should call handleChange when input is changed', () => {
    const component = shallow(
      <LoginForm
        handleChange={global.handleChange}
        handleSubmit={global.handleSubmit}
        response={global.response}
      />
    )

    component.find('Input[name="email"]').simulate('change')
    component.find('Input[name="password"]').simulate('change')

    expect(global.handleChange.calledTwice).toEqual(true)
  })

  it('should show error response', () => {
    const response = {
      hasFailed: true,
      response: {
        message: 'Something is wrong.'
      }
    }
    const component = shallow(
      <LoginForm
        handleChange={global.handleChange}
        handleSubmit={global.handleSubmit}
        response={response}
      />
    )

    const errorMessage = component.find('Message')

    expect(errorMessage.props().hidden).toEqual(false)
    expect(errorMessage.find('p').text()).toEqual(response.response.message)
  })

})
