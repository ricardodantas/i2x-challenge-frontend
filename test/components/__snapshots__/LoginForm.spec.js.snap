// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render 1`] = `
<div
  className="ui container ui raised very padded text container segment"
>
  <h2>
    Login
  </h2>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui input"
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui input"
      >
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
      </div>
    </div>
    <button
      className="ui primary button"
      onClick={[Function]}
      tabIndex={undefined}
      type="submit"
    >
      Sign In
    </button>
  </form>
  <div
    className="ui hidden negative message"
  >
    <div
      className="header"
    >
      Ops...
    </div>
    <p />
  </div>
</div>
`;
